documentation/src/docs/wire-protocol.md

```diff
@@ -9,104 +9,105 @@ You may obtain a copy of the License at
-3.  Suggested Length of Reply - int (4 bytes)
-    1.  This is how much data the client wants. They won't necessarily get that much.
 
-### Segment Read - Reply
+## Segment Read - Reply
 
-1.  Segment that was read - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
-2.  Offset that was read from - Long (8 bytes)
-3.  Is at Tail - Boolean (1 bit)
-4.  Is at EndOfSegment - (1 bit)
-5.  Data - Binary (remaining length in message)
+1.  Segment that was read: A length of 2 byte String, followed by that many bytes of Java's Modified UTF-8).
+2.  Offset that was read from: 8 byte Long.
+3.  Is at Tail: 1 bit Boolean.
+4.  Is at `EndOfSegment`: 1 bit.
+5.  Data: Binary (remaining length in the message).
 
-The client requests to read from a particular stream at a particular offset, it then receives one or more replies in the form of SegmentRead messages. These contain the data they requested (assuming it exists). The server may decided to give the client more or less data than it asked for, in as many replies as it sees fit.
+The client requests to read from a particular Segment at a particular Offset. It then receives one or more replies in the form of `SegmentRead` messages. These contain the data they requested (assuming it exists). The server decides on more or less data to give the client than it asked for, in as many replies as it sees fit.
 
-Appending
----------
+# Appending
 
-### Setup Append - Request
+## Setup Append - Request
+1.  `RequestId`: 8 byte Long. (This field contains the client-generated ID that has been propagated to identify a client request).
+2.  `writerId`: UUID (16 bytes) Identifies this appender.
+3.  Segment to append: A length of 2 byte String, followed by that many bytes of Java's Modified UTF-8.
+4. `delegationToken`: This was added to perform _auth_. It is an opaque-to-the-client token provided by the Controller that says it's allowed to make this call.
 
-1.  ConnectionId - UUID (16 bytes) Identifies this appender.
-2.  Segment to append to. - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
+## Append Setup - Reply
 
-### Append Setup - Reply
+1.  `RequestId`: 8 byte Long. (This field contains the client-generated ID that has been propagated to identify a client request).
+2.  Segment that can be appended: A length of 2 byte String, followed by that many bytes of Java's Modified UTF-8.
+3.  Final `writerId`: UUID (16 bytes) Identifies the requesting appender.
+4.  Final  `lastEventNumber`: 8 byte Long.
 
-1.  Segment that can be appended to. - String (2 byte length, followed by that many bytes of Java's Modified UTF-8)
-2.  ConnectionId - UUID (16 bytes) Identifies the requesting appender.
-3.  ConnectionOffsetAckLevel - Long (8 bytes) What was the last offset received and stored on this segment for this connectionId (0 if new)
+## AppendBlock - Request
 
-### BeginAppendBlock - Request 
+1. Final `writerId`: (UUID 16 bytes).
```

Comment:
In previous messages there is not `Final` before the field name. If this refers to the Java keyword, it think that it could be removed.

Reply:
Removed